packages/stan/src/test/derived.js

```diff
@@ -0,0 +1,169 @@
+		const unsubscribe = registry.subscribe( sum, () => {} );
+		await flushImmediatesAndTicks( 2 );
+		expect( registry.get( sum ) ).toEqual( 21 );
+		unsubscribe();
+	} );
+
+	it( 'should only compute derived atoms when they have subscribers or when you try to retrieve their value', () => {
+		const mock = jest.fn();
+		mock.mockImplementation( () => 10 );
+		const count1 = createAtom( 1 );
+		const count2 = createAtom( 1 );
+		const sum = createDerivedAtom(
+			( { get } ) => get( count1 ) + get( count2 ) + mock()
+		);
+		const registry = createAtomRegistry();
+		// Creating an atom or adding it to the registry don't trigger its resolution
+		expect( mock ).not.toHaveBeenCalled();
+		expect( registry.get( sum ) ).toEqual( 12 );
+		// Calling "get" triggers a resolution.
+		expect( mock ).toHaveBeenCalledTimes( 1 );
+
+		// This shouldn't trigger the resolution because the atom has no listener.
+		registry.set( count1, 2 );
+		expect( mock ).toHaveBeenCalledTimes( 1 );
+
+		// Subscribing triggers the resolution again.
+		const unsubscribe = registry.subscribe( sum, () => {} );
+		expect( mock ).toHaveBeenCalledTimes( 2 );
+		expect( registry.get( sum ) ).toEqual( 13 );
+		unsubscribe();
+	} );
+
+	it( 'should notify subscribers on change', () => {
+		const count1 = createAtom( 1 );
+		const count2 = createAtom( 1 );
+		const sum = createDerivedAtom(
+			( { get } ) => get( count1 ) + get( count2 )
+		);
+		const registry = createAtomRegistry();
+		const listener = jest.fn();
+		const unsubscribe = registry.subscribe( sum, listener );
+
+		registry.set( count1, 2 );
+		expect( listener ).toHaveBeenCalledTimes( 1 );
+
+		registry.set( count2, 2 );
+		expect( listener ).toHaveBeenCalledTimes( 2 );
+
+		unsubscribe();
+	} );
+} );
+
+describe( 'updating derived atoms', () => {
+	it( 'should allow derived atoms to update dependencies', () => {
+		const count1 = createAtom( 1 );
+		const count2 = createAtom( 1 );
+		const sum = createDerivedAtom(
+			( { get } ) => get( count1 ) + get( count2 ),
+			( { set }, value ) => {
+				set( count1, value / 2 );
+				set( count2, value / 2 );
+			}
+		);
+		const registry = createAtomRegistry();
+		registry.set( sum, 4 );
+		expect( registry.get( count1 ) ).toEqual( 2 );
+		expect( registry.get( count2 ) ).toEqual( 2 );
+	} );
+
+	it( 'should allow nested derived atoms to update dependencies', () => {
+		const count1 = createAtom( 1 );
+		const count2 = createAtom( 1 );
+		const sum = createDerivedAtom(
+			( { get } ) => get( count1 ) + get( count2 ),
+			( { set }, value ) => {
+				set( count1, value / 2 );
+				set( count2, value / 2 );
+			}
+		);
+		const multiply = createDerivedAtom(
+			( { get } ) => get( sum ) * 3,
+			( { set }, value ) => {
+				set( sum, value / 3 );
+			}
+		);
+		const registry = createAtomRegistry();
+		registry.set( multiply, 18 );
+		expect( registry.get( count1 ) ).toEqual( 3 );
+		expect( registry.get( count2 ) ).toEqual( 3 );
+	} );
+} );
```

Comment:
Those tests are great for documentation purposes and help better undesirable how it all should work ❤️